src/openms/source/CHEMISTRY/RNaseDigestion.cpp

```diff
@@ -100,31 +144,80 @@ namespace OpenMS
     }
     else // proper enzyme cleavage
     {
-      vector<Size> fragment_pos(1, 0);
+      // with variable A->I, would enzyme cut after/before I, but not unmod. A?
+      bool cuts_after_I = (variable_inosine_ &&
+                           boost::regex_search("I", cuts_after_regex_) &&
+                           !boost::regex_search("A", cuts_after_regex_));
+      bool cuts_before_I = (variable_inosine_ &&
+                            boost::regex_search("I", cuts_before_regex_) &&
+                            !boost::regex_search("A", cuts_before_regex_));
+      // list of fragment (start) positions, and whether cleavage there would
+      // require a variable modification (A->I) before or after the site:
+      vector<tuple<Size, bool, bool>> fragment_pos;
+      fragment_pos.emplace_back(0, false, false);
       for (Size i = 1; i < rna.size(); ++i)
       {
-        if (boost::regex_search(rna[i - 1]->getCode(), cuts_after_regex_) &&
-            boost::regex_search(rna[i]->getCode(), cuts_before_regex_))
+        // standard case:
+        bool cuts_after_regular =
+          boost::regex_search(rna[i - 1]->getCode(), cuts_after_regex_);
+        bool cuts_before_regular =
+          boost::regex_search(rna[i]->getCode(), cuts_before_regex_);
+        if (cuts_after_regular && cuts_before_regular)
+        {
+          fragment_pos.emplace_back(i, false, false);
+          continue;
+        }
+        // cases involving variable modifications:
+        bool cuts_after_special = (!cuts_after_regular && cuts_after_I &&
+                                   (rna[i - 1]->getCode() == "A"));
+        if (cuts_after_special && cuts_before_regular)
         {
-          fragment_pos.push_back(i);
+          fragment_pos.emplace_back(i, true, false);
+          continue;
+        }
+        bool cuts_before_special = (!cuts_before_regular && cuts_before_I &&
+                                    (rna[i]->getCode() == "A"));
+
+        if (cuts_after_regular && cuts_before_special)
+        {
+          fragment_pos.emplace_back(i, false, true);
+          continue;
+        }
+        if (cuts_after_special && cuts_before_special)
+        {
+          fragment_pos.emplace_back(i, true, true);
+          continue;
         }
       }
-      fragment_pos.push_back(rna.size());
+      fragment_pos.emplace_back(rna.size(), false, false);
```

Comment:
can you add a inline comment what the false, false stand for?

Reply:
Done.